sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusModelFactory.cs

```diff
@@ -310,7 +353,27 @@ public static class ServiceBusModelFactory
             };
 
         /// <summary>
-        /// Creates a new <see cref="RuleProperties"/> instance for mocking.
+        /// Creates a new <see cref="Azure.Messaging.ServiceBus.Administration.SubscriptionProperties"/> instance for mocking.
+        /// </summary>
+        public static SubscriptionProperties SubscriptionProperties(
```

Comment:
Interesting.. I'm curious why we need factory methods that take the options bag rather than the individual properties of SubscriptionProperties.

Reply:
It's definitely not necessary, it just made the code for some of the tests a lot simpler, like [this one](https://github.com/Azure/azure-sdk-for-net/blob/2787a86b05336e79dc00afff46d12d70384cf1a7/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Samples/Sample15_MockingClientTypes.cs#L316). Otherwise, you have to break up the options into each piece anyway ([this is what it looked like before](https://github.com/Azure/azure-sdk-for-net/blob/97f52e5e4bbde7f92c42e0b6a4e59ecf77e9e56d/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Samples/Sample15_MockingClientTypes.cs#L316))

Reply:
Hmm I'm not sure that users will actually need the properties to match in practice. More likely they are trying to test some application code but they need to mock the call to the Service Bus library.